src/grid/grid.component.ts

```diff
@@ -116,6 +125,17 @@ export class IgxGridComponent implements OnInit, OnDestroy, AfterContentInit, Af
         this.cdr.markForCheck();
     }
 
+    get filteredData() {
+        return this._filteredData;
+    }
+
+    set filteredData(value) {
+        if (this.rowSelectable) {
+            this._filteredData = value;
```

Comment:
Why is the `_filteredData` set only when there is row selection? The update of the header checkbox should happen on that condition, but the setter should work regardless.